src/views/components/omnibar.jsx

```diff
@@ -13,16 +15,29 @@ export default class Omnibar extends React.Component {
 
     changeQuery(e) {
         let regex = new RegExp("^[a-zA-Z0-9]*$");
+        let regexOnlyNums = new RegExp("^[0-9]*$");
         if(regex.test(e.target.value)) {
-            if (e.target.value.length === 6) {
+            try {
+                let result = readAddress(e.target.value);
                 this.setState({
                     query: ''
                 });
-                OmnibarController.getStudent(e.target.value);
-            } else {
-                this.setState({
-                    query: e.target.value
-                });
+                if(result.type === 'model') {
+                    ModelController.getModelAndItems(e.target.value);
+                } else if (result.type === 'item') {
+                    OmnibarController.displayItem(e.target.value);
+                }
+            } catch (f) {
+                if (e.target.value.length === 6 && regexOnlyNums.test(e.target.value)) {
```

Comment:
Remember in SE1021 when you were told that Exceptions are only for exceptional behavior, not meant for typical program flow? Entering a student ID in the omnibar isn't that exceptional.

My only solution is:
```
if (e.target.value.length === 6 && regexOnlyNums.test(e.target.value)) {
  // show student
} else {
  // do all the model/item stuff
}
```